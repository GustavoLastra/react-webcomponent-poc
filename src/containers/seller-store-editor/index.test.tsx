const mockLocaleSelector = jest.fn();
jest.mock('../language-provider/selector', () => ({
  __esModule: true,
  localeSelector: mockLocaleSelector
}));

import React from 'react';
import ConnectedSellerStoreEditor, { SellerStoreEditor } from './index';
import ReactTestRenderer from 'react-test-renderer/shallow';
import configureStore from 'redux-mock-store';

const mockStore = configureStore([]);

describe('<SellerStoreEditor />', () => {
  afterEach(() => {
    mockLocaleSelector.mockReset();
  });

  it('connected component should render and match the snapshot', () => {
    mockLocaleSelector.mockReturnValue('de');
    // @ts-ignore
    const component = <ConnectedSellerStoreEditor store={mockStore()} />;
    const renderer = ReactTestRenderer.createRenderer();
    renderer.render(component);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it('container should render and match the snapshot', () => {
    const props: any = {
      locale: 'tr',
      setLocale: () => undefined
    };
    const component = <SellerStoreEditor {...props} />;
    const renderer = ReactTestRenderer.createRenderer();
    renderer.render(component);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
