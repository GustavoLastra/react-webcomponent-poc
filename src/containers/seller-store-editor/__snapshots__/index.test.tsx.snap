// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SellerStoreEditor /> connected component should render and match the snapshot 1`] = `
<Context.Provider
  value={null}
>
  <Unknown
    locale="de"
    setLocale={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />
</Context.Provider>
`;

exports[`<SellerStoreEditor /> container should render and match the snapshot 1`] = `
<SellerStoreEditor>
  <Switch>
    <Route
      path="/counter"
    >
      <Memo(Connect(Component)) />
    </Route>
    <Route
      path="/"
    >
      <Memo(Connect(IPCheck)) />
    </Route>
  </Switch>
  <button
    onClick={[Function]}
  >
    Locale: 
    tr
  </button>
</SellerStoreEditor>
`;
